### PR TITLE
Connect linalg on tensors to rest of the CPU side.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
Connecting the code generated from Linalg on tensors to rest of the
CPU side requires setting the tile size (that is dynamic when tile +
distribute happens in flow level) to the static tile size that is
computed using LaunchConfig.

### CODE REVIEW

File: iree/compiler/Conversion/Common/Transforms.cpp [hunk 1]
```diff
@@ -219,52 +219,48 @@ LogicalResult getLinalgOps(FuncOp funcOp,
 namespace {
 static size_t kMaxHALDimensions = 3;
 
-/// Sets the flow.dispatch.workgroup_size operation to the constant value passed
+/// Sets the hal.dispatch.workgroup.size operation to the constant value passed
```
Comment (benvanik) [inline iree/compiler/Conversion/Common/Transforms.cpp:222]: hal.interface.workgroup.size

**Review by benvanik — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToLLVM/LinalgTileAndDistributePass.cpp [hunk 2]
```diff
@@ -142,8 +142,18 @@ void LinalgTileAndDistributePass::runOnOperation() {
         return signalPassFailure();
       }
     } else {
-      if (failed(materializeStaticLaunchInformation(funcOp, launchConfig,
-                                                    tiledLoops.size()))) {
+      SmallVector<int64_t, 4> defaultWorkloadPerWorkgroup(tiledLoops.size(), 1);
+      if (!defaultWorkloadPerWorkgroup.empty())
+        defaultWorkloadPerWorkgroup.back() = 4;
```
Comment (hanhanW) [inline iree/compiler/Conversion/LinalgToLLVM/LinalgTileAndDistributePass.cpp:147]: Wrap with braces

File: iree/compiler/Dialect/Flow/Transforms/test/dispatch_linalg_on_tensors_dynamic.mlir [hunk 3]
```diff
@@ -42,3 +42,46 @@ func @tensor(%arg0 : tensor<?x?xf32>, %arg1 : tensor<?x?xf32>,
...
+//      CHECK:       scf.for %[[IV1:.[a-zA-Z0-9_]+]]
+//      CHECK:       %[[V1:.+]] = flow.dispatch.input.load %[[ARG4]]
+//      CHECK:       %[[V2:.+]] = flow.dispatch.input.load %[[ARG5]]
+//      CHECK:       %[[D0:.+]] = affine.min #[[MAP1]](%[[ARG2]], %[[IV0]], %[[WG_SIZE_Y]])
+//      CHECK:       %[[D1:.+]] = affine.min #[[MAP1]](%[[ARG3]], %[[IV1]], %[[WG_SIZE_X]])
+//      CHECK:       %[[INIT:.+]] = linalg.init_tensor [%[[D0]], %[[D1]]]
+//      CHECK:       %[[RESULT:.+]] = linalg.generic
+// CHECK-SAME:         ins(%[[V1]], %[[V2]] : tensor<?x?xf32>, tensor<?xf32>)
+// CHECK-SAME:         outs(%[[INIT]] : tensor<?x?xf32>)
+//      CHECK:         flow.dispatch.output.store %[[RESULT]], %[[ARG6]], offsets = [%[[IV0]], %[[IV1]]], sizes = [%[[D0]], %[[D1]]]
```
Comment (hanhanW) [inline iree/compiler/Dialect/Flow/Transforms/test/dispatch_linalg_on_tensors_dynamic.mlir:87]: Add a newline at the end of file

File: iree/test/e2e/linalg_tensor_ops/BUILD [hunk 4]
```diff
@@ -27,8 +27,8 @@ package(
 )
 
 iree_check_single_backend_test_suite(
-    name = "check_llvm-ir_llvm",
-    srcs = glob(["*.mlir"]),
+    name = "check_matmul_llvm-ir_llvm",
+    srcs = [glob(["*.mlir"])],
```
Comment (hanhanW) [inline iree/test/e2e/linalg_tensor_ops/BUILD:31]: revert, ie remove `[]`?

File: iree/test/e2e/linalg_tensor_ops/add.mlir [hunk 5]
```diff
@@ -0,0 +1,35 @@
+func @tensor() attributes { iree.module.export } {
```
Comment (hanhanW) [inline iree/test/e2e/linalg_tensor_ops/add.mlir:1]: You miss command to test this file.

**Review by hanhanW — APPROVED** [review]

**Pull request merged**